File: Leetcode/Review/20211003/reversePrint.go
package _0211003

func reversePrint(head *ListNode) []int {
	res:=[]int{}
	for head!=nil{
		res=append(res,head.Val)
		head=head.Next
	}
	for i:=0;i<len(res)/2;i++{
		res[i],res[len(res)-i-1]=res[len(res)-i-1],res[i]
	}
	return res
}

File: Leetcode/剑指offer/easy/20210913/reversePrint.cpp
class Solution {
public:
    vector<int> reversePrint(ListNode* head) {
        vector<int> res;
        while(head!=NULL){
            res.push_back(head->val);
            head=head->next;
        }
        for(int i=0;i<res.size()/2;i++){
            swap(res[i],res[res.size()-i-1]);
        }
        return res;
    }
};
